cachito/workers/celery_logging.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import inspect
+import logging
+
+from celery.utils.log import ColorFormatter
+
+
+class RequestFilter(logging.Filter):
+    """A log filter that sets ``request_id`` on the log record."""
+
+    def __init__(self, request_id, *args, **kwargs):
+        """
+        Initialize a filter that sets ``request_id`` on the log record.
+
+        :param request_id: the request ID to set on the record
+        """
+        super().__init__(*args, **kwargs)
+        self._request_id = request_id
+
+    def filter(self, record):
+        """
+        Set ``request_id`` on the log record.
+
+        :param logging.LogRecord record: the log record
+        :return: always returns ``True`` so the log is not filtered out
+        :rtype: bool
+        """
+        record.request_id = self._request_id
+        return True
+
+
+def _get_function_arg_value(arg_name, func, args, kwargs):
+    """
+    Get the value of the given argument name.
+
+    :param str arg_name: the name of the argument to get
+    :param function func: the function the arguments are for
+    :param list args: the list of arguments passed to the function
+    :param dict kwargs: the dictionary or keyword arguments passed to the function
+    :return: the argument value or ``None``
+    """
+    argspec = inspect.getfullargspec(func).args
```

Comment:
Don't we need to handle the case where the actual function is decorated multiple times? I don't think accessing just the immediate `func.__wrapped__` is sufficient. It may work now, but as soon as we decorate it with another decorator it's going to fail.

```python
original_func = func                                
while getattr(func, '__wrapped__', None):           
    original_func = func.__wrapped__                
argspec = inspect.getfullargspec(original_func).args

```

Reply:
You're right, but the suggestion you proposed is an infinite loop in Cachito. I filed a PR to fix this in IIB:
https://github.com/release-engineering/iib/pull/133

Reply:
Thanks for catching this!